public/src/managers/ServerProvider.ts

```diff
@@ -112,8 +112,12 @@ export class ServerProvider implements IUserProvider, ICourseProvider {
     }
 
     public async changeUserState(link: Enrollment, state: Enrollment.UserStatus): Promise<Status> {
+        const originalState = link.getStatus();
```

Comment:
Why are these variables called `link` and `state`? (I assume it is legacy reasons)
Would it be reasonable to name these as `enrollment` and `enrollStatus` or something along those lines?

Reply:
Yeah, absolutely. These were indeed the legacy names for those fields.